src/Sarif/Autogenerated/FileData.cs

```diff
@@ -56,6 +56,12 @@ public SarifNodeKind SarifNodeKind
         [DataMember(Name = "length", IsRequired = false, EmitDefaultValue = false)]
         public int Length { get; set; }
 
+        /// <summary>
+        /// The role or roles played by the file in the analysis.
+        /// </summary>
+        [DataMember(Name = "roles", IsRequired = false, EmitDefaultValue = false)]
+        public FileRole Roles { get; set; }
```

Comment:
Sorry I wasn't clear. `file.roles` is an array, each of whose elements is a member of the `FileRoles` enum. So it should generate `public IList<FileRole> Roles; { get; set; }`. Did you change the schema to make this singular?

Reply:
Oh I thought it was a Flags enum

Reply:
I'm sorry, you're right!

Reply:
We need tests for serializing/deserializing this property. I suppose just four tests: (serializing, deserializing) x (one enum value, multiple enum values) = 4.

Reply:
It looks like there is a problem with the way flags enums are generated. Because the members are not given explicit values, they get the sequential defaults {0, 1, 2, 3, 4, 5 ...} instead of the bitfield power of 2 values.

Reply:
As we discussed: merge as is (with two tests ignored), fix JSchema code gen, update SDK to new JSchema, re-enable tests.